internal/node/controlapi.go

```diff
@@ -41,7 +41,7 @@ func NewApiListener(log *slog.Logger, mgr *WorkloadManager, node *Node) *ApiList
 	efftags := config.Tags
 	efftags[controlapi.TagOS] = runtime.GOOS
 	efftags[controlapi.TagArch] = runtime.GOARCH
-	efftags[controlapi.TagCPUs] = strconv.FormatInt(int64(runtime.NumCPU()), 10)
+	efftags[controlapi.TagCPUs] = strconv.FormatInt(int64(runtime.GOMAXPROCS(0)), 10)
```

Comment:
Based on the docs, `GOMAXPROCS(0)` does not change the value that is otherwise returned by `NumCPU()` and `GOMAXPROCS(0)` will eventually be deprecated. I suggest reverting back to `NumCPU`.

Reply:
This actually doesn't change anything meaningful, this is only changing the value of the CPU tag that the node reports upon query.

Reply:
Using `runtime.NumCPU()` is actually a bug-- it returns the number "available" on the machine, not the number `GOMAXPROCS`. Using `runtime.GOMAXPROCS(0)` actually returns the number of CPUs configured via `GOMAXPROCS`.

Reply:
We could alternatively parse it out of `os.Getenv`.

Reply:
> We could alternatively parse it out of os.Getenv

This will only work if GOMAXPROCS is overridden by environment

Reply:
For some reason I was thinking the process set `GOMAXPROCS` if otherwise unset. But that makes sense.